pkg/security/rules/engine.go

```diff
@@ -518,6 +518,11 @@ func (e *RuleEngine) SetRulesetLoadedCallback(cb func(es *rules.RuleSet, err *mu
 
 // HandleEvent is called by the probe when an event arrives from the kernel
 func (e *RuleEngine) HandleEvent(event *model.Event) {
+	// no process context, let's skip
+	if event.ProcessContext == nil {
+		return
+	}
+
 	// don't eval event originating from myself
 	if !e.probe.Opts.DontDiscardRuntime && event.ProcessContext.Pid == e.pid {
```

Comment:
why not 
if !e.probe.Opts.DontDiscardRuntime && event.ProcessContext != nil && event.ProcessContext.Pid == e.pid

Reply:
we could do that yes